packages/turbo-workspace-convert/__tests__/managers/pnpm.test.ts

```diff
@@ -0,0 +1,246 @@
+      async (from, shouldThrow) => {
+        const { root, tmpDirectory } = useFixture({
+          fixture: `../${from}/basic`,
+        });
+
+        const read = async () => pnpm.read({ workspaceRoot: path.join(root) });
+        if (shouldThrow) {
+          expect(read).rejects.toThrow("Not a pnpm workspaces project");
+          return;
+        }
+        const project = await pnpm.read({
+          workspaceRoot: path.join(root),
+        });
+        expect(project.name).toEqual("pnpm-workspaces");
+        expect(project.packageManager).toEqual("pnpm");
+        // paths
+        expect(project.paths.root).toMatch(
+          new RegExp(`^.*pnpm\/${tmpDirectory}$`)
+        );
+        expect(project.paths.lockfile).toMatch(
+          new RegExp(`^.*pnpm\/${tmpDirectory}\/pnpm-lock.yaml$`)
+        );
+        expect(project.paths.packageJson).toMatch(
+          new RegExp(`^.*pnpm\/${tmpDirectory}\/package.json$`)
+        );
+        // workspaceData
+        expect(project.workspaceData.globs).toEqual(["apps/*", "packages/*"]);
+        expect(project.workspaceData.workspaces).toHaveLength(4);
+        project.workspaceData.workspaces.forEach((workspace) =>
+          validateWorkspace(tmpDirectory, workspace)
+        );
+      }
+    );
+  });
+
+  describe("convertLock", () => {
+    test.each<[PackageManagers]>([["npm"], ["yarn"], ["pnpm"]])(
+      "converts %s workspaces lockfile to pnpm lockfile",
+      async (from) => {
+        const { root } = useFixture({ fixture: `../${from}/basic` });
+        const project = await MANAGERS[from].read({
+          workspaceRoot: root,
+        });
+        const interactive = false;
+        const dry = false;
+        await pnpm.create({
+          project,
+          to: { name: "pnpm", version: "7.27.0" },
+          logger: new Logger({ interactive, dry }),
+          options: {
+            interactive,
+            dry,
+          },
+        });
+
+        expect(existsSync(project.paths.lockfile)).toBe(true);
+
+        await pnpm.convertLock({
+          project,
+          logger: new Logger(),
+          options: {
+            interactive,
+            dry,
+          },
+        });
+
+        expect(
+          existsSync(path.join(project.paths.root, "pnpm-lock.yaml"))
+        ).toBe(true);
+      }
+    );
+
+    test("fails gracefully when lockfile is missing", async () => {
+      const { root } = useFixture({ fixture: `../npm/basic` });
+      const project = await MANAGERS.npm.read({
+        workspaceRoot: root,
+      });
+      fs.rmSync(project.paths.lockfile);
+      await pnpm.convertLock({
+        project,
+        logger: new Logger(),
+        options: {
+          interactive: false,
+          dry: false,
+        },
+      });
+      expect(existsSync(path.join(project.paths.root, "pnpm-lock.yaml"))).toBe(
+        false
+      );
+    });
+  });
+
+  describe("clean", () => {
+    test.each([
+      [true, true],
```

Comment:
named keys in an object would be easier to read

Reply:
Agreed. Jest does support something similar to this, but it's pretty buggy from what I found (the name replacement wasn't working - and there's a few issues about it as well)